packages/fuchsia_ctl/lib/src/ssh_client.dart

```diff
@@ -92,21 +97,84 @@ class SshClient {
   Future<OperationResult> runCommand(String targetIp,
       {@required String identityFilePath,
       @required List<String> command,
-      Duration timeoutMs = defaultSshTimeoutMs}) async {
+      Duration timeoutMs = defaultSshTimeoutMs,
+      String logFilePath,
+      FileSystem fs}) async {
     assert(targetIp != null);
     assert(identityFilePath != null);
     assert(command != null);
 
-    return OperationResult.fromProcessResult(
-      await processManager
-          .run(
-            getSshArguments(
-              identityFilePath: identityFilePath,
-              targetIp: targetIp,
-              command: command,
-            ),
-          )
-          .timeout(timeoutMs),
+    final bool logToFile = !(logFilePath == null || logFilePath.isEmpty);
+    FileSystem fileSystem = fs ?? const LocalFileSystem();
+    IOSink logFile;
+    Logger logger;
+
+    // If no file is passed to this method we create a memoryfile to keep to
+    // return the stdout in OperationResult.
+    if (logToFile) {
+      fileSystem.file(logFilePath).existsSync() ??
+          fileSystem.file(logFilePath).deleteSync();
+      fileSystem.file(logFilePath).createSync();
+      final IOSink data = fileSystem.file(logFilePath).openWrite();
+      logger = PrintLogger(out: data);
+    } else {
+      fileSystem = MemoryFileSystem();
+      fileSystem.file('logs')..createSync();
+      logFile = fileSystem.file('logs').openWrite();
+      logger = PrintLogger();
+    }
+
+    final Process process = await processManager.start(
+      getSshArguments(
+        identityFilePath: identityFilePath,
+        targetIp: targetIp,
+        command: command,
+      ),
     );
+    final StreamSubscription<String> stdoutSubscription = process.stdout
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .listen((String log) {
+      if (!logToFile) {
+        logFile.writeln(log);
+      } else {
+        logger.info(log);
+      }
```

Comment:
Why can't we just do `logger.info(log)` here? Won't it write to the file if it's attached to the IOSink, and otherwise jut write to stdout?

Reply:
Done. It was implemented like that because logfile adds datetime and log level and all the ssh commands in fuchsia_ctl expects output with no additional prepended.